### ISSUE TITLE
Different color palettes per color property

### ISSUE THREAD

**Issue by nosolosw** [WordPress/gutenberg#25613]
In reading the [TwentyTwentyOne announcement](https://make.wordpress.org/core/2020/09/23/introducing-twenty-twenty-one/) and looking at the proposal for colors:

![Mobile-Colors](https://user-images.githubusercontent.com/583546/94140512-a3209a00-fe6b-11ea-883c-d3802d85d84e.png)

this paragraph stood out to me:

> The theme also uses a limited color palette: a pastel green background color, and two shades of dark grey for text.

Via theme.json we already have the ability to have different color palettes per block (for example, to allow only a subset of colors in the paragraph block). However, the color palette is shared across all the color properties (background, text color, and link color).

Would it be useful if we allowed theme authors to create contextual palettes? I'm thinking that one approach could be to allow having background & foreground palettes (includes text + link color). An alternative could be to allow different color palettes for each color property.

**Comment by nosolosw** [comment]
cc @melchoyce @jffng @kjellr @mtias @youknowriad @jorgefilipecosta for feedback.

**Comment by aristath** [comment]
YES!!!
Having different palettes for color & background would be amazing!
Usually text-color only needs 2 options, white/black to accomodate dark/light backgrounds. Backgrounds on the other hand need to be a bit more versatile, so having separate palettes would make a lot of sense

**Comment by mtias** [comment]
There are two ways to go about this:

- Block specific palettes (like Table does, I believe) that a theme can customize
- Open up the system for multiple palettes anywhere

I think the first one is a bit awkward and doesn't necessarily lead to a better experience. But supporting multiple palettes seems much better to me in general. Right now if a theme wants to register a limited palette with some primary colors (i.e. 2020) all the other colors from core disappear. That might be desirable in some cases, but in others it makes it really difficult to pick a standard "blue" if the theme didn't include one. If you could instead "swap" between palettes we could keep core's set of main colors + the theme + user colors. We have discussed registering a more vivid set of colors for core, which could benefit from the ability to have different palettes registered.

To this specific suggestion, I don't think we should have palettes "per role" (background, foreground, border, etc). It seems unnecessary complexity.

We have discussed template area specific palettes as something to support for advanced cases. Blocks could always definie their own contexts, so in theme.json a theme could specify a specific palette for block x. but I also think this can get too complicated very quickly and I prefer exploring the multiple palettes first and see if that can be enough, both being more flexible and more clear UX wise.

**Comment by florianbrinkmann** [comment]
+1 for the possibility to add different color palettes per role. Maybe it could be something like a opt-in to use for developers, so if only a `settings.color.palette` is specified, that would create the same result like now, but it is also possible to use `settings.color.textPalette` (or something like that) to use a different palette for the text color.

**Comment by landwire** [comment]
I think it is crucial to be able to have the same colour palettes across all blocks that support it. If you design/specify a specific colour theme (background, text and link colours) for a client, you really want to register it in one place and not on all possible blocks.
I also think the differentiation per role (background, text, link, whatever else there might be) is necessary, as there will be quite different colours used and if they are all muddled together it is a confusing UI for the user.
Often we have 3 text colours, but about 6 background colours. Link colours we don't work with, so the ability to hide certain palettes would also be necessary.

So in short:
1. Register colour palettes in one place (global)
2. Ability to have different colour palettes per role (background, text, link, etc)
3. Ability to remove certain colour palettes
4. Being able to override global colour settings inside blocks

**Comment by oandregal** [comment]
I'm going to close this one. There're no plans to implement this and we are going to ship in WordPress 5.8 (and already in the plugin) the ability to use core colors in addition to theme ones.
